v2/AdminCore.WebApp/Controllers/HolidayController.cs

```diff
@@ -22,74 +27,184 @@ namespace AdminCore.WebApi.Controllers
     public IActionResult GetHolidayByEmployeeId(int employeeId)
     {
-      return Ok();
+      ResponseMessage<IList<EventDto>> holidaysDtos = _holidayEventService.GetEventsByEmployeeId(employeeId);
+      var holidays = _mapper.Map<ResponseMessage<IList<EventDto>>, List<HolidayViewModel>>(holidaysDtos);
+      return Ok(holidays);
     }
 
     [HttpPost]
     public IActionResult CreateHoliday(CreateHolidayViewModel viewModel)
     {
-      return Ok();
+      IList<string> errors = ValidateCreateHolidayViewModel(viewModel);
+      return CreateResponseBasedOnErrors(errors, viewModel);
     }
 
     [HttpPut]
-    public IActionResult UpdateHoliday(UpdateHolidayViewModel viewModel)
+    public IActionResult UpdateHoliday(UpdateHolidayViewModel updateHoliday)
     {
-      return Ok();
+      var eventDto = _mapper.Map<UpdateHolidayViewModel, EventDto>(updateHoliday);
+      return Ok(_holidayEventService.UpdateEvent(eventDto));
     }
 
     [HttpPut("approveHoliday")]
     public IActionResult ApproveHoliday(ApproveHolidayViewModel approveHoliday)
     {
-      return Ok();
+      var eventDto = _mapper.Map<ApproveHolidayViewModel, EventDto>(approveHoliday);
+      return Ok(_holidayEventService.ApproveEvent(eventDto));
     }
 
     [HttpPut("cancelHoliday")]
     public IActionResult CancelHoliday(CancelHolidayViewModel cancelHoliday)
     {
-      return Ok();
+      var eventDto = _mapper.Map<CancelHolidayViewModel, EventDto>(cancelHoliday);
+      return Ok(_holidayEventService.CancelEvent(eventDto));
     }
 
     [HttpPut("rejectHoliday")]
     public IActionResult RejectHoliday(RejectHolidayViewModel rejectHoliday)
     {
-      return Ok();
+      var responses = _holidayEventService.RejectEvent(
+        rejectHoliday.EventId,
+        rejectHoliday.Message,
+        int.Parse(_authenticatedUser.RetrieveUserId()));
+      return Ok(responses);
     }
 
-    [HttpGet("findByDateBetween/{eventDate}")]
-    public IActionResult GetHolidayByDateBetween(EventDate eventDate)
+    [HttpGet("findByDateBetween/{startDate}/{endDate}")]
+    public IActionResult GetHolidayByDateBetween(string startDate, string endDate)
     {
-      return Ok();
+      if (IsDatesValid(startDate, endDate))
+      {
+        return Ok(GetEventsBetweenDates(startDate, endDate));
+      }
+      return BadRequest("Please use a valid date format for start and/or end dates");
     }
 
     [HttpGet("findByHolidayStatus/{holidayStatusId}")]
-    public IActionResult GetHolidayByStatus(int holidayStatusId)
+    public IActionResult GetHolidayByStatusType(int holidayStatusId)
+    {
+      var holidaysDtos = _holidayEventService.GetByStatusType((EventStatuses)holidayStatusId, EventTypes.AnnualLeave);
+      var holidays = _mapper.Map<ResponseMessage<IList<EventDto>>, List<HolidayViewModel>>(holidaysDtos);
+
+      return Ok(holidays);
+    }
+
+    protected bool ValidateDate(string date)
+    {
+      try
+      {
+        DateTime dt = DateTime.Parse(date);
```

Comment:
Very valid way of checking parsing a date. c# 6 has a new way which is pretty nice. 
var dateToParse = Date.parse(date, var out dateToParse);
if (dateToParse != null)
 // etc etc

Reply:
DateTime dateTime = null;
        if (DateTime.TryParse("2019-01-01", out dateTime))
        {
            Console.WriteLine(dateTime);
        }

this way if it throws an exception, you can still catch it.